### DISCUSSION TITLE
Pub/Sub Telemetry

### DISCUSSION THREAD

**Discussion by haseebH** [open-telemetry/opentelemetry-go#1884] [Q&A]
### Description

I am new to OpenTelmetry. I am trying to implement OpenTelmetry for a pub/sub application. Right now, I am trying to pass parent trace and span info from Publisher to the subscriber. But there is no way available at the moment to create a context with parent information. Is there is any way I can solve this problem?

**Comment by Aneurysm9** [comment]
Please see the [context propagation system](https://pkg.go.dev/go.opentelemetry.io/otel@v0.20.0/propagation), which you should use to communicate trace context information between processes.  The [Sarama instrumentation](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/instrumentation/github.com/Shopify/sarama/otelsarama/message.go) can perhaps be used as a guide for how to use the context propagation system with message headers in such a system.
